src/vstest.console/CommandLine/AssemblyMetadataProvider.cs

```diff
@@ -103,17 +103,28 @@ private Architecture GetArchitectureFromAssemblyMetadata(string path)
             switch (peReader.PEHeaders.CoffHeader.Machine)
             {
                 case Machine.Amd64:
-                    return Architecture.X64;
                 case Machine.IA64:
                     return Architecture.X64;
                 case Machine.Arm64:
                     return Architecture.ARM64;
                 case Machine.Arm:
                     return Architecture.ARM;
                 case Machine.I386:
-                    return Architecture.X86;
+                    // If requires 32bit or if R2R
```

Comment:
It is not clear to me what is the reasoning there (I would also spell out Ready to run at least once in the comment.

Reply:
We can distinguish AnyCPU only from the set of `CorFlags.Requires32Bit`, but in case of Ready to Run image that flag is not "updated" and ignored, if so we check it the module is IL only or not if it's IL only it means that is a R2R and we're already in the correct architecture x86. In all other cases the architecture will end inside the correct switch branch

Reply:
```suggestion
                    // We can distinguish AnyCPU only from the set of CorFlags.Requires32Bit, but in case of Ready                                         
                    // to Run image that flag is not "updated" and ignored. So we check if the module is IL only or                    
                    // not. If it's IL only it means that is a R2R (Ready to Run) and we're already in the correct architecture x86. In all                    
                    // other cases the architecture will end inside the correct switch branch.
```